kickoff/static/suggestions.js

```diff
@@ -189,52 +189,37 @@ function populatePartial(productName, version, previousBuilds, partialElement) {
     // Check that all partials match a build.
     partialConsistencyCheck(partialsADIVersion, previousReleases);
 
-    partial = [];
-    partialAdded = 0;
-
     // When we have the ADI for Firefox Beta or Thunderbird, we can remove
     // this special case
-    if (isTB(productName) || isFxBeta) {
-        // No ADI, select the three first
-        partial = addLastVersionAsPartial(version, previousReleases, 3);
-        partialAdded = 3;
-        partialElement.val(partial.join());
-        return true;
-    } else {
+    partials = isTB(productName) || isFxBeta ?
+        craftLastReleasesAsPartials(version, previousReleases, 3) :
         // The first partial will always be the previous published release
-        partial = addLastVersionAsPartial(version, previousReleases, 1);
-        partialAdded++;
-    }
+        craftLastReleasesAsPartials(version, previousReleases, 1);
 
     // Firefox X.0 releases are published to the beta channel, so generate a partial
     // from the most recent beta build to speed that up
-    if (isFirefox(productName) && version.match(/^\d+\.0$/)) {
+    if (partials.length < maximumNumberOfPartials && isFirefox(productName) && version.match(/^\d+\.0$/)) {
```

Comment:
It looks like we'll end up with 1 beta and 2 older release builds when we do RC builds with this change (currently we have 1 beta and 3 release builds). We don't have a test for this scenario, so could you add one and see if the code changes result in a different result ?

Reply:
I agree with you about tests, I should add one to validate my assumption of the fix.

Regarding the rest, I think it should stay the same.

From the history in the production environment, 57.0 is the only release that shipped with 4 partials (1 beta and 3 releases). That was a special requirement which allowed automation to generate bz2 updates for 55.0.3 (and lower). If I remember correctly, relman added 55.0.3 by hand.

Past releases had 3 partials (1 beta and 2 older releases). The current behavior of the prod env goes that way. I kept the best for the end: test case covers it https://github.com/mozilla-releng/ship-it/blob/21ae4fde5df440edbe4edcc1645a1f99a3e39655/kickoff/test/js/tests.js#L233 \o/

Reply:
Ah, you're quite right. I was getting confused with our new freedom to have more than 3 partials.